core/trino-main/src/main/java/io/trino/sql/planner/EffectivePredicateExtractor.java

```diff
@@ -390,6 +393,9 @@ public Expression visitValues(ValuesNode node, Void context)
                             }
                             else {
                                 Type type = types.get(node.getOutputSymbols().get(i));
+                                if (hasNestedNulls(type, item)) {
```

Comment:
Forgot to mention it before, but please add a comment explaining that this is a workaround to deal with the fact that array and row comparisons don't currently support null elements.